lib/perso.ml

```diff
@@ -3599,6 +3631,71 @@ and eval_str_person_field conf base env ((p, p_auth) as ep) = function
       match get_env "p_link" env with
       | Vbool _ -> null_val
       | _ -> get_iper p |> string_of_iper |> Mutil.encode |> safe_val)
+  (* carrousel functions *)
+  | "keydir" -> Image.default_portrait_filename base p |> str_val
+  | "keydir_img_nbr" ->
+      string_of_int (List.length (Image.get_keydir_files conf base p))
+      |> str_val
+  | "keydir_old_img_nbr" ->
+      string_of_int (List.length (Image.get_keydir_old_files conf base p))
+      |> str_val
+  | "keydir_img_notes" -> (
+      match get_env "keydir_img" env with
+      | Vstring f ->
+          let ext = Filename.extension f in
+          let fname = Filename.chop_suffix f ext in
+          Image.get_keydir_img_notes conf base p fname |> str_val
+      | _ -> raise Not_found)
+  | "keydir_img_src" -> (
+      match get_env "keydir_img" env with
+      | Vstring f -> (
+          let ext = Filename.extension f in
+          let fname = Filename.chop_suffix f ext in
+          let n = Image.get_keydir_img_notes conf base p fname in
+          match String.index_opt n '\n' with
+          | Some i -> (
+              let s1 =
+                if String.length n > i then
+                  String.sub n (i + 1) (String.length n - i - 1)
+                else ""
+              in
+              match String.index_opt s1 '\n' with
+              | Some j -> String.sub s1 0 j |> str_val
+              | None -> "" |> str_val)
```

Comment:
I don't understand this, `Image.get_keydir_img_notes conf base p fname` is a string with the note of the image fname?
Or is it a list of notes delimited by `\n`?

Why do you have to take a substring starting and ending with `\n`?